app/controllers/application_controller.rb

```diff
@@ -10,11 +10,24 @@ class ApplicationController < ActionController::Base
   helper LinksHelper
   helper_method :current_user
   helper_method :sso_identity
+  helper_method :current_estate
 
   def current_user
     sso_identity&.user
   end
 
+  def current_estate
+    return unless sso_identity
+    @_current_estate ||= begin
+      estate_id = session[:current_estate]
```

Comment:
I think you're right – I need to additionally check that the estate is still accessible.